### PR TITLE
Translation field & more

### LINKED ISSUE #123: Create a translation-field Web component

### PR DESCRIPTION
**Pull request by jswale**
Fixes #123

### CODE REVIEW

File: src/components/molecules/p6-translation/locales/p6-translation.i18n.fr.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "missingDefaultTranslation": "Traduction requise pour la langue par défaut \"#{name} (#{code})\"",
+  "missingTranslation": "Traduction absente pour la langue \"#{languages}\"",
+  "addTooltip": "Ajouter une traduction",
+  "deleteTooltip": "Supprimer la translation"
```
Comment (vuongc) [inline src/components/molecules/p6-translation/locales/p6-translation.i18n.fr.json:5]: `translation` by `traduction`

File: src/components/atoms/p6-language/p6-language.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+  /**
+   * Marks as read only.
+   */
+  @Prop({ attribute: "readOnly" }) public readOnly = false;
+
+  /**
+   *
+   */
+  @Event() p6Change!: EventEmitter<ValidEventDetail<P6LanguageValue>>;
```
Comment (vincentv) [inline src/components/atoms/p6-language/p6-language.tsx:69]: a more verbose comment could be useful for the documentation

File: src/components/atoms/p6-select-native/p6-select-native.tsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+  /**
+   *
+   */
+  @Event() p6FormUnregister!: EventEmitter<P6SelectControl>;
+
+  /**
+   *
+   */
+  @Event() p6Change!: EventEmitter<ValidEventDetail<P6SelectValue>>;
```
Comment (vincentv) [inline src/components/atoms/p6-select-native/p6-select-native.tsx:84]: more verbose comments could be useful for the documentation

File: src/components/molecules/p6-translation/p6-translation-line/p6-translation-line.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+  /**
+   *
+   */
+  @Event() p6KeyChange!: EventEmitter<ValidEventDetail<LanguageCode>>;
+
+  /**
+   *
+   */
+  @Event() p6ValueChange!: EventEmitter<ValidEventDetail<string>>;
```
Comment (vincentv) [inline src/components/molecules/p6-translation/p6-translation-line/p6-translation-line.tsx:71]: more verbose comments could be useful for the documentation

File: src/components/molecules/p6-translation/p6-translation.tsx [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+
+  /**
+   *
+   */
+  @Event() p6FormRegister!: EventEmitter<P6TranslationControl>;
+
+  /**
+   *
+   */
+  @Event() p6FormUnregister!: EventEmitter<P6TranslationControl>;
```
Comment (vincentv) [inline src/components/molecules/p6-translation/p6-translation.tsx:82]: more verbose comments could be useful for the documentation

File: src/components/molecules/p6-translation/p6-translation.tsx [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+
+    return Promise.resolve(isValid);
+  }
+
+  render(): JSX.Element {
+    return (
+      <Host>
+        <p6-field>
+          <p6-label slot="label">
+            Description
```
Comment (vincentv) [inline src/components/molecules/p6-translation/p6-translation.tsx:154]: we should use the localization here.

**Pull request merged**